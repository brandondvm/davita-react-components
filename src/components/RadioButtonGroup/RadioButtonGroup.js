import React, { useState } from 'react';
import PropTypes from 'prop-types';

import RadioButton from '../RadioButton/RadioButton';

const RadioButtonGroup = ({ options = [], label, register, watch, watchFields = [], errors, onRadioButtonChange }) => {
  const [ radioOptions, setRadioOptions] = useState(options);

  const optionClick = radioOption => {
    onRadioButtonChange && onRadioButtonChange(radioOption.name, radioOption.value);
    setRadioOptions(options.map(option => ({
      ...option,
      checked: option.id === radioOption.id
    })));
  }

  return (
    <div className="radiobutton-group-component">
      { label && <p className="radiobutton-group-component_label">{ label }</p> }
      {
        radioOptions.map((option) => {
          const { label, name, id, value, checked, textInput, radioText, validation } = option;          
          return (
            <RadioButton 
              key={ id }
              inputLabel={ label } 
              inputName={ name } 
              inputId={ id }  
              inputValue={ value } 
              isInputChecked={ checked }  
              onOptionClick={() => optionClick(option)}
              register={register}
              validation={validation}
              errors={errors}
              textInput={textInput}
              radioText={radioText}
              watchFields={watchFields}
              watch={watch} />
          );
        })
      }
    </div>
  );
};

RadioButtonGroup.propTypes = {
  options: PropTypes.array,
  register: PropTypes.func,
  watchFields: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.array,
  ]),
};

export default RadioButtonGroup;
