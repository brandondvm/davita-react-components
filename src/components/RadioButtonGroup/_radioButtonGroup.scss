.radiobutton-group-component {
  &_label {
    color: $onyx;
    font-family: $regular-font;
    font-size: 14px;
  }
}
