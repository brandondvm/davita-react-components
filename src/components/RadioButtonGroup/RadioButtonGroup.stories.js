import React from 'react';

import RadioButtonGroup from './RadioButtonGroup';

const radioGroupItems = [
  { id: 'radioOption1', label: 'Option 1', name: 'radiobuttongroup', value: 1, checked: false },
  { id: 'radioOption2', label: 'Option 2', name: 'radiobuttongroup', value: 2, checked: false },
];

export default {
  title: 'RadioButton',
  component: RadioButtonGroup,
};

const Template = (args) => <div className="storybook-wrapper"><RadioButtonGroup {...args} /></div>;

export const Default = Template.bind({});
Default.args = {
  options: radioGroupItems
};
