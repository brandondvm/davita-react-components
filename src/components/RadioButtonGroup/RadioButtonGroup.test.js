import enzyme from 'enzyme';
import React from 'react';

import RadioButtonGroup from './RadioButtonGroup';

const mockProps = {
  options: [
    {
      id: "radiobutton1",
      label: "radiobutton 1",
      name: "radioButtonGroup",
      value: "radiobutton 1",
      checked: false,
    },
    {
      id: "radiobutton2",
      label: "radiobutton 2",
      name: "radioButtonGroup",
      value: "radiobutton 2",
      checked: false,
    },
    {
      id: "radiobutton3",
      label: "radiobutton 3",
      name: "radioButtonGroup",
      value: "radiobutton 3",
      checked: false,
    },
  ],
  onItemClick: jest.fn(),
};

function setup(passedProps = {}) {
  const updatedProps = {
    ...mockProps,
    ...passedProps,
  };

  return enzyme.shallow(<RadioButtonGroup { ...updatedProps } />);
}

describe('RadioButtonGroup Component', () => {
  test('Component renders correctly', () => {
    const wrapper = setup();
    expect(wrapper.get(0)).toMatchSnapshot();
  });

  test('Component should render 3 radio buttons correctly', () => {
    const wrapper = setup();
    const radiobuttonsElements = wrapper.find('RadioButton');
    expect(radiobuttonsElements.length).toEqual(3);
  });
});
