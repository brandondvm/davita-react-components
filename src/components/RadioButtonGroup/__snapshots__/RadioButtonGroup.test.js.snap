// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtonGroup Component Component renders correctly 1`] = `
<div
  className="radiobutton-group-component"
>
  <RadioButton
    inputId="radiobutton1"
    inputLabel="radiobutton 1"
    inputName="radioButtonGroup"
    inputValue="radiobutton 1"
    isInputChecked={false}
    onOptionClick={[Function]}
    watchFields={Array []}
  />
  <RadioButton
    inputId="radiobutton2"
    inputLabel="radiobutton 2"
    inputName="radioButtonGroup"
    inputValue="radiobutton 2"
    isInputChecked={false}
    onOptionClick={[Function]}
    watchFields={Array []}
  />
  <RadioButton
    inputId="radiobutton3"
    inputLabel="radiobutton 3"
    inputName="radioButtonGroup"
    inputValue="radiobutton 3"
    isInputChecked={false}
    onOptionClick={[Function]}
    watchFields={Array []}
  />
</div>
`;
